crypto/dso/dso_dlfcn.c

```diff
@@ -359,18 +363,33 @@ static int dladdr(void *ptr, Dl_info *dl)
             || ((addr >= (uintptr_t)this_ldi->ldinfo_dataorg)
                 && (addr < ((uintptr_t)this_ldi->ldinfo_dataorg +
                             this_ldi->ldinfo_datasize)))) {
+            char *buffer, *member;
+            size_t buffer_sz, member_len;
+            buffer_sz = strlen(this_ldi->ldinfo_filename) + 1;
+            member = (char *)((uintptr_t)this_ldi->ldinfo_filename +
```

Comment:
` (uintptr_t) ` cast looks weird to me .
Are AiX using wide char or some explanatory comment is missing.

Reply:
Right, there is inconsistency in how it's used. I mean it's passed to strlen, but here it's casted. But to answer question, no, it's not wide char or anything, and one can actually just write `this_ldi->ldinfo_filename + buffer_sz` without any casts whatsoever. Could you fix it, Matthias, for better readability? And while you are at it, could you add even empty line Rich is talking about... And line 339 is too long... It should be possible to omit `(struct ld_info *)` cast there, because OPENSSL_malloc returns one-size-fits-all void \*.

Reply:
For the last two points, yes, I can do that. But the (uintptr_t) cast is necessary, as otherwise the compiler doesn't get the offset right. Hm ... looking at it again, it might be a left over. I'll check that too.

Reply:
What I see on 7.1 is that ldinfo_filename is declared as char, so no cast should be necessary. And if it's needed on some older[?] version, then `(char *)` would do just fine.

Reply:
I see : i forget about operator precedence, offsetting is done before cast.